### ISSUE TITLE
subclasses of Loader must provide a get_template_sources() method

### ISSUE THREAD

**Issue by kakulukia** [matannoam/pypugjs#25]
I just upgraded to Django 2 and am getting the above error

**Comment by kakulukia** [comment]
but its weird .. git blame on the django loaders base.py where this is happening tells me that the file was last touched in February .. so whats actually causing the error?

**Comment by timgraham** [comment]
Updates are required to remove usage of deprecation features. See [Template loader APIs have changed](https://docs.djangoproject.com/en/dev/releases/1.9/#template-loader-apis-have-changed). 

The relevant note in the 2.0 release notes is "The load_template and load_template_sources template loader methods are removed.". If you turn on deprecation warnings in Django 1.11, you'll get a helpful tip.

**Comment by kakulukia** [comment]
oh .. i didnt realize that deprecation warnings are disabled by default now and was wondering that i never saw one. o.0

Thanks for the tip!

**Comment by kakulukia** [comment]
@timgraham is there any possibility to let deprecaton warnings be enabled through the use of some settings? I liked to see them popup without specifically asking for them, because otherwise nobody will notice until its too late. (like above)
I'd like to have an option other than needing to type the extra "python -Wd" before manage.py.
Is that possible?

**Comment by timgraham** [comment]
Yes, you can use Python's `warnings` module. For example, https://github.com/django/django/blob/244cc401559e924355cf943b6b8e66ccf2f6da3a/tests/runtests.py#L23-L28.

**Comment by danqing** [comment]
@kakulukia wonder if you can publish your repo to pip (maybe under `pypug`) and we can go from there. This repo seems to have followed pyjade and become abandoned :(

**Comment by kakulukia** [comment]
Im thinking about it :)
